esphome/components/logger/logger.h

```diff
@@ -178,7 +181,8 @@ class Logger : public Component {
                                                         va_list args, char *buffer, uint16_t *buffer_at,
                                                         uint16_t buffer_size) {
 #if defined(USE_ESP32) || defined(USE_LIBRETINY) || defined(USE_ZEPHYR)
-    this->write_header_to_buffer_(level, tag, line, this->get_thread_name_(), buffer, buffer_at, buffer_size);
+    char buff[MAX_POINTER_REPRESENTATION];
+    this->write_header_to_buffer_(level, tag, line, this->get_thread_name_(buff), buffer, buffer_at, buffer_size);
```

Comment:
```suggestion
    #ifdef USE_ZEPHYR
      char buff[MAX_POINTER_REPRESENTATION];
      this->write_header_to_buffer_(level, tag, line, this->get_thread_name_(buff), buffer, buffer_at, buffer_size);
    #else
      this->write_header_to_buffer_(level, tag, line, this->get_thread_name_(), buffer, buffer_at, buffer_size);
    #endif
```
Given this function is marked HOT and we know logger performance can block the system to the point we get failures in issue reports, I'd probably branch it

Reply:
done